### ISSUE TITLE
bug(build): can't build the project

### ISSUE THREAD

**Issue by magrinj** [prescottprue/react-redux-firebase#808]
Hi there,

I'm actually trying to add `reauthenticate` functions to the project because I need them in my current project, but I can't `build` or `link` the project locally.

(Using `login` function instead of `reauthenticate` actually is not working as it clear the state, en then signIn, so the user is logout during a small amount of time)

I'm using node v13.2.0

The beginning of the stack trace looks like, something related to `grpc` apparently:

```
error /Users/magrin_j/Library/Caches/Yarn/v6/.tmp/bce6b65d749ad1f8ed9c7f5f1e365449.30be85f1cce54fbf7ca6bd05876ef727d76568ce.prepare/node_modules/grpc: Command failed.
Exit code: 1
Command: node-pre-gyp install --fallback-to-build --library=static_library
Arguments: 
Directory: /Users/magrin_j/Library/Caches/Yarn/v6/.tmp/bce6b65d749ad1f8ed9c7f5f1e365449.30be85f1cce54fbf7ca6bd05876ef727d76568ce.prepare/node_modules/grpc
Output:
node-pre-gyp info it worked if it ends with ok
node-pre-gyp info using node-pre-gyp@0.13.0
node-pre-gyp info using node@13.2.0 | darwin | x64
node-pre-gyp WARN Using request for node-pre-gyp https download 
node-pre-gyp info check checked for "/Users/magrin_j/Library/Caches/Yarn/v6/.tmp/bce6b65d749ad1f8ed9c7f5f1e365449.30be85f1cce54fbf7ca6bd05876ef727d76568ce.prepare/node_modules/grpc/src/node/extension_binary/node-v79-darwin-x64-unknown/grpc_node.node" (not found)
node-pre-gyp http GET https://node-precompiled-binaries.grpc.io/grpc/v1.24.0/node-v79-darwin-x64-unknown.tar.gz
node-pre-gyp http 404 https://node-precompiled-binaries.grpc.io/grpc/v1.24.0/node-v79-darwin-x64-unknown.tar.gz
node-pre-gyp WARN Tried to download(404): https://node-precompiled-binaries.grpc.io/grpc/v1.24.0/node-v79-darwin-x64-unknown.tar.gz 
node-pre-gyp WARN Pre-built binaries not found for grpc@1.24.0 and node@13.2.0 (node-v79 ABI, unknown) (falling back to source compile with node-gyp) 
node-pre-gyp http 404 status code downloading tarball https://node-precompiled-binaries.grpc.io/grpc/v1.24.0/node-v79-darwin-x64-unknown.tar.gz 
gyp info it worked if it ends with ok
gyp info using node-gyp@3.6.3
gyp info using node@13.2.0 | darwin | x64
gyp info ok 
gyp info it worked if it ends with ok
gyp info using node-gyp@3.6.3
gyp info using node@13.2.0 | darwin | x64
gyp info spawn /usr/bin/python
gyp info spawn args [
gyp info spawn args   '/Users/magrin_j/Library/Caches/Yarn/v6/.tmp/bce6b65d749ad1f8ed9c7f5f1e365449.30be85f1cce54fbf7ca6bd05876ef727d76568ce.prepare/node_modules/node-gyp/gyp/gyp_main.py',
gyp info spawn args   'binding.gyp',
gyp info spawn args   '-f',
gyp info spawn args   'make',
gyp info spawn args   '-I',
gyp info spawn args   '/Users/magrin_j/Library/Caches/Yarn/v6/.tmp/bce6b65d749ad1f8ed9c7f5f1e365449.30be85f1cce54fbf7ca6bd05876ef727d76568ce.prepare/node_modules/grpc/build/config.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/Users/magrin_j/Library/Caches/Yarn/v6/.tmp/bce6b65d749ad1f8ed9c7f5f1e365449.30be85f1cce54fbf7ca6bd05876ef727d76568ce.prepare/node_modules/node-gyp/addon.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/Users/magrin_j/.node-gyp/13.2.0/include/node/common.gypi',
gyp info spawn args   '-Dlibrary=shared_library',
gyp info spawn args   '-Dvisibility=default',
gyp info spawn args   '-Dnode_root_dir=/Users/magrin_j/.node-gyp/13.2.0',
gyp info spawn args   '-Dnode_gyp_dir=/Users/magrin_j/Library/Caches/Yarn/v6/.tmp/bce6b65d749ad1f8ed9c7f5f1e365449.30be85f1cce54fbf7ca6bd05876ef727d76568ce.prepare/node_modules/node-gyp',
gyp info spawn args   '-Dnode_lib_file=/Users/magrin_j/.node-gyp/13.2.0/<(target_arch)/node.lib',
gyp info spawn args   '-Dmodule_root_dir=/Users/magrin_j/Library/Caches/Yarn/v6/.tmp/bce6b65d749ad1f8ed9c7f5f1e365449.30be85f1cce54fbf7ca6bd05876ef727d76568ce.prepare/node_modules/grpc',
gyp info spawn args   '-Dnode_engine=v8',
gyp info spawn args   '--depth=.',
gyp info spawn args   '--no-parallel',
gyp info spawn args   '--generator-output',
gyp info spawn args   'build',
gyp info spawn args   '-Goutput_dir=.'
gyp info spawn args ]
gyp info ok 
gyp info it worked if it ends with ok
gyp info using node-gyp@3.6.3
gyp info using node@13.2.0 | darwin | x64
gyp info spawn make
gyp info spawn args [ 'BUILDTYPE=Release', '-C', 'build' ]
  CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/surface/init.o
In file included from ../deps/grpc/src/core/lib/surface/init.cc:29:
In file included from ../deps/grpc/src/core/lib/channel/channel_stack.h:44:
In file included from ../deps/grpc/src/core/lib/debug/trace.h:27:
In file included from ../deps/grpc/src/core/lib/gprpp/global_config.h:92:
In file included from ../deps/grpc/src/core/lib/gprpp/global_config_env.h:24:
In file included from ../deps/grpc/src/core/lib/gprpp/global_config_generic.h:24:
../deps/grpc/src/core/lib/gprpp/memory.h:116:56: warning: unused parameter 'hint' [-Wunused-parameter]
```

I've found some issue about grpc `1.24.0` and some people are saying that using gprc `1.23.3` should fix this issue, I'm not really sure if it's the problem right there

**Comment by prescottprue** [comment]
Could you provide your package.json so I can try to replicate on my end? Is this on Windows or Mac/Linux? Do you see the same thing when trying to build in CI? Or when trying to use a tool like Codesandbox?

**Comment by magrinj** [comment]
@prescottprue Actually it happen, on the fork of the project, the package.json is exactly the same as right know, it seams that the error isn't happening on CI, because my last PR pass all the tests
I'm using Mac OS X

**Comment by prescottprue** [comment]
Did you try using node 12 or node 10? Wondering if one of the dependencies is what is causing it since I'm not aware of anything not being compatible with newer node version - I'll look into try to repro.

**Comment by magrinj** [comment]
@prescottprue Sounds it works with node 10, not with node 13 and node 8, maybe something should be add in the package.json like:

```
  "engines": {
    "node": "10"
  },
```

**Comment by prescottprue** [comment]
Updating firebase (dev dependency) to `^7.6.2` fixed the issue with node 13 (grpc is a dependency of it), so released that in v3.0.7. Reach out if it doesn't fix things for you - thanks for reporting!

**Comment by starcrest** [comment]
Seeing the same problem with node v14 in the material example. Error pasted below.

Works* when I use node@12.

Thanks!

```
2 errors generated.
make: *** [Release/obj.target/grpc_node/ext/call.o] Error 1
gyp ERR! build error
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:194:23)
gyp ERR! stack     at ChildProcess.emit (events.js:314:20)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:276:12)
gyp ERR! System Darwin 19.4.0
gyp ERR! command "/usr/local/Cellar/node/14.5.0/bin/node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "build" "--fallback-to-build" "--library=static_library" "--module=/Users/starcrest/sw/gcp/react-redux-firebase/examples/complete/material/node_modules/grpc/src/node/extension_binary/node-v83-darwin-x64-unknown/grpc_node.node" "--module_name=grpc_node" "--module_path=/Users/starcrest/sw/gcp/react-redux-firebase/examples/complete/material/node_modules/grpc/src/node/extension_binary/node-v83-darwin-x64-unknown" "--napi_version=6" "--node_abi_napi=napi" "--napi_build_version=0" "--node_napi_label=node-v83"
gyp ERR! cwd /Users/starcrest/sw/gcp/react-redux-firebase/examples/complete/material/node_modules/grpc
gyp ERR! node -v v14.5.0
gyp ERR! node-gyp -v v5.1.0
gyp ERR! not ok
node-pre-gyp ERR! build error
node-pre-gyp ERR! stack Error: Failed to execute '/usr/local/Cellar/node/14.5.0/bin/node /usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js build --fallback-to-build --library=static_library --module=/Users/starcrest/sw/gcp/react-redux-firebase/examples/complete/material/node_modules/grpc/src/node/extension_binary/node-v83-darwin-x64-unknown/grpc_node.node --module_name=grpc_node --module_path=/Users/starcrest/sw/gcp/react-redux-firebase/examples/complete/material/node_modules/grpc/src/node/extension_binary/node-v83-darwin-x64-unknown --napi_version=6 --node_abi_napi=napi --napi_build_version=0 --node_napi_label=node-v83' (1)
node-pre-gyp ERR! stack     at ChildProcess.<anonymous> (/Users/starcrest/sw/gcp/react-redux-firebase/examples/complete/material/node_modules/grpc/node_modules/node-pre-gyp/lib/util/compile.js:83:29)
node-pre-gyp ERR! stack     at ChildProcess.emit (events.js:314:20)
node-pre-gyp ERR! stack     at maybeClose (internal/child_process.js:1051:16)
node-pre-gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:287:5)
node-pre-gyp ERR! System Darwin 19.4.0
```

**Comment by starcrest** [comment]
*Also, separately, I  think you need redux-firestore in the package.json. Pasting the line below works for me:
```
+    "redux-firestore": "^0.7.3",
```

**Comment by prescottprue** [comment]
@starcrest Thanks for reaching out, feel free to make a PR if you get a chance

**Comment by sdwru** [comment]
> Updating firebase (dev dependency) to `^7.6.2` fixed the issue with node 13 (grpc is a dependency of it), so released that in v3.0.7. Reach out if it doesn't fix things for you - thanks for reporting!

This worked for me with node.js `v14.5`
